user-service/handlers/user_handler.go

```diff
@@ -31,6 +32,43 @@ func NewUserHandler(s services.UserService, v validator.Validate, rts services.R
 	return &h
 }
 
+func (h *UserHandler) HandleChangePassword(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	var req dtos.ChangePasswordDto
+	if ok, err := requests.ReadAndValidateJson(w, h.v, r.Body, &req); !ok {
+		if err != nil {
+			log.Printf("failed to process change password request: %v", err)
+		}
+		return
+	}
+
+	err := h.s.ChangePassword(r.Context(), &req)
+
+	switch {
+	case errors.Is(err, services.ErrInvalidCurrentPassword):
+		_ = respond.Unauthorized(w, "Invalid current password.")
+		return
+
+	case errors.Is(err, services.ErrTooFrequentPasswordChange):
+		_ = respond.BadRequest(w, "Password changed too frequently.")
+		return
+	case err != nil:
+		_ = respond.InternalServerError(w)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
+
+	b := map[string]any{
+		"message": "Password changed successfully",
+	}
+
+	if err := json.NewEncoder(w).Encode(b); err != nil {
+		_ = respond.InternalServerError(w)
+		return
+	}
```

Comment:
Use `respond.Ok` here. It handles response messaage standarization, HTTP status code, encoding and error handling.

Make sure to use full English sentences in messages, including the peroids.

```suggestion

	if err := respond.Ok(w, "Password changed successfully."); err != nil {
		log.Printf("failed to write change password response: %v", err)
	}
```